Deduplicated code, eliminated manual array<->collection transfers.

tkroman opened the pull request:
// Unrelated to this pull-request:
@mateiz, I also took liberty to uncomment that line here: b21cbe688fdf5136b7d2569b1edc19fea117a8cc. Let me know if you consider it worth separate pull-request or should we wait?

UPD: Now it looks even more clearly duplicated for me, maybe I should try to generify it even more?

tkroman committed:
Deduplicated code, eliminated manual array<->collection transfers.

AmplabJenkins commented:
Thank you for your pull request. An admin will review this request soon.

andyk commented:
Jenkins, test this please.

AmplabJenkins commented:
Thank you for submitting this pull request.

All automated tests for this request have passed.

Refer to this link for build results: http://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/426/

tkroman committed:
Experimental generification (for testing purposes)

tkroman committed:
Fixed comment (in case this commits are gonna pass type tests).

rxin commented:
@JoshRosen can you take a look at this?

JoshRosen commented:
After a cursory look, this seems fine to me; I think the existing tests in https://github.com/mesos/spark/blob/f649dabb4a3a57cb25a852808297fb718cbfedd4/core/src/test/scala/spark/JavaAPISuite.java#L81 should cover this.

tkroman commented:
So, maybe someone could order Jenkins to test this?

mateiz commented:
Jenkins, retest this please

AmplabJenkins commented:
Thank you for submitting this pull request. 

Unfortunately, the automated tests for this request have failed.

Refer to this link for build results: http://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/470/

mateiz commented:
Jenkins, retest this please

mateiz commented:
Jenkins had some trouble yesterday but I'll test it manually if it doesn't work.

AmplabJenkins commented:
Thank you for submitting this pull request. 

Unfortunately, the automated tests for this request have failed.

Refer to this link for build results: http://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/471/

tkroman committed:
Temporary workaround to see if it fixes the issue.

If it works as desired, I'm going to make it more clear.
Currently problem is that constructors for not- RDD[T]-instances take additional arguments (classManifests), so I need to figure out whether it's possible to generify that.

tkroman committed:
fixed typo

AmplabJenkins commented:
Thank you for your pull request. An admin will review this request soon.

mateiz commented:
Jenkins, retest this please

mateiz commented on core/src/main/scala/spark/api/java/JavaSparkContext.scala:
```diff
@@ -258,27 +258,39 @@ class JavaSparkContext(val sc: SparkContext) extends JavaSparkContextVarargsWork
+  // override def union(first: JavaDoubleRDD, rest: java.util.List[JavaDoubleRDD]): JavaDoubleRDD = {
+  //   val rdds: Seq[RDD[Double]] = (Seq(first) ++ asScalaBuffer(rest)).map(_.srdd)
+  //   new JavaDoubleRDD(sc.union(rdds))
+  // }
```
Actually, I do have one comment on the code: please don't leave commented-out lines in the project. If you don't need these versions of union(), delete them.

AmplabJenkins commented:
Thank you for submitting this pull request. 

Unfortunately, the automated tests for this request have failed.

Refer to this link for build results: http://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/480/
